Hello! My hypothesis was that the Perspective would mark capitalized content, content that contained the negative context, and content that contained some type of threat/allusion to death as more toxic than other content that did not contain those things. To get more specific I would say that I thought that Perspective would mark content with allusion to harm as the most toxic.  I had three sets of sentences where I varied those factors. The first set was a more "positive" sentence that I then modified by adding either a curse word, capitalizing the sentence, adding a reference to harm, or a combination of such. The final sentence of each set was a combination of a curse word, reference to harm, and capitalization. I repeated this with a more "negative" base sentence as well as a sentence that was more vague. 

In [59]:
import requests
import json
API_KEY = "my-api-key"
TOXICITY_API_URL = "https://commentanalyzer.googleapis.com/v1alpha1/comments:analyze"

comments = [
    "I loved your opinion in this article",
    "I fucking loved your opinion in this article.",
    "I FUCKING LOVED YOUR OPINION IN THIS ARTICLE",
    "I loved your opinion in this article you should go kill yourself",
    "I fucking loved your opinion in this article you should go kill yourself",
    "I FUCKING LOVED YOUR OPINION IN THIS ARTICLE YOU SHOULD GO KILL YOURSELF",
    
    "I hated your opinion in this article",
    "I fucking hated your opinion in this article.",
    "I FUCKING HATED YOUR OPINION IN THIS ARTICLE",
    "I hated your opinion in this article you should go kill yourself",
    "I fucking hated your opinion in this article you should go kill yourself",
    "I FUCKING HATED YOUR OPINION IN THIS ARTICLE YOU SHOULD GO KILL YOURSELF",
    
    "Wow this was crazy",
    "WOW THIS WAS CRAZY",
    "Wow this was crazy you should die",
    "Wow this was crazy you should fucking die",
    "WOW THIS WAS CRAZY YOU SHOULD FUCKING DIE",

]

def get_toxicity_score(comment):
    data = {
        "comment": {"text": comment},
        "requestedAttributes": {"TOXICITY": {}},
        "doNotStore": True
 }
    params = {"key":"my-api-key"}
    response = requests.post(TOXICITY_API_URL, params=params, json=data)
    response_dict = json.loads(response.content)
    print(response_dict)  # Print response for debugging
    return response_dict["attributeScores"]["TOXICITY"]["summaryScore"]["value"]

for comment in comments:
    score = get_toxicity_score(comment)
    print(f"Comment: {comment}\nToxicity score: {score}\n")




{'attributeScores': {'TOXICITY': {'spanScores': [{'begin': 0, 'end': 36, 'score': {'value': 0.014953613, 'type': 'PROBABILITY'}}], 'summaryScore': {'value': 0.014953613, 'type': 'PROBABILITY'}}}, 'languages': ['en'], 'detectedLanguages': ['en']}
Comment: I loved your opinion in this article
Toxicity score: 0.014953613

{'attributeScores': {'TOXICITY': {'spanScores': [{'begin': 0, 'end': 45, 'score': {'value': 0.45761138, 'type': 'PROBABILITY'}}], 'summaryScore': {'value': 0.45761138, 'type': 'PROBABILITY'}}}, 'languages': ['en'], 'detectedLanguages': ['en']}
Comment: I fucking loved your opinion in this article.
Toxicity score: 0.45761138

{'attributeScores': {'TOXICITY': {'spanScores': [{'begin': 0, 'end': 44, 'score': {'value': 0.51980776, 'type': 'PROBABILITY'}}], 'summaryScore': {'value': 0.51980776, 'type': 'PROBABILITY'}}}, 'languages': ['en'], 'detectedLanguages': ['en']}
Comment: I FUCKING LOVED YOUR OPINION IN THIS ARTICLE
Toxicity score: 0.51980776

{'attributeScores': {'TOXI

From the tests, it seemed that Perspective would indeed increase the toxicity score when changes like capitalization, allusion to harm, negative context, and curse words were added. In the positive "I loved your opinion in this article" sentence example, the normal sentence had a toxicity score of 0.014953613 but spiked when I added a curse word (0.45761138) and then rose even more when I capitalized all of it (0.51980776). When I only added the reference to harm to the base sentence, it also rose a substansial amount (to 0.8540474), and only rose higher when I added a curse word (to 0.93383175) and then again when I capitalized the entire sentence (to 0.944597). 

This pattern of rising toxicity scores with these additions to the base sentence was also present when I started with the other base sentences. With the "I hated your opinion in this article" section, since it was a initial negative context, the toxicity score in this base sentence was higher as well (0.50789946 compared to the 0.014953613) and therefore the resulting additions had higer toxicity scores as well. 

The addition of a reference to harm caused the biggest jump in toxicity score (when only adding one change) in all 3 different base sentences, with the addition of a curse word (in this case "fucking") as the second biggest spike, and then capitalization as the next. 

I think that Perspectives performance in this test shows some bias in regard to deciding that capitalized sentences are more toxic than uncapitalized ones even though the content is the same. In some online communities, capitalization of whole sentences can be used in as a form of excitement/extra positiveness, but Perspective does not seem to think so. I think it may see excessive capitalization as aggressive and therefore assign a higher score to it. It also seemed to decide that the addition of a curse word even in a positive context like, "I fucking loved your opinion in this article", is more toxic which is perhaps not always the case. Curse words could be used in a non-toxic, excited way as well I guess. 

Overall, Perspective assigned toxicity scores in the way I thought they would, with the references to harm as the addition that would spike the toxicity score the most. It was interesting seeing how much "toxicity value" they assigned each new addition to the sentences though.